### PR TITLE
Use "command priority" of merlin-mode to handle racy situations

### PR DESCRIPTION
**Pull request by def-lkb**
The fix for handling competing access to merlin.
The idea is that `flycheck-ocaml` indicates its low priority and if merlin can't satisfy the request, merlin-cancelled will be signaled (only `merlin-sync-to-point` can fail).

Note the `(defvar merlin-command-priority)`: the file is lexically-scoped, but merlin-mode expects this setting to be dynamically redefined. Using a defvar seems to be the normal way to recover dynamic scoping.

### CODE REVIEW

File: flycheck-ocaml.el [hunk 1]
```diff
@@ -39,6 +39,8 @@
 (require 'merlin)
 (require 'flycheck)
 
+(defvar merlin-command-priority) ;; enjoy dynamic scoping
```
Comment (lunaryorn) [inline flycheck-ocaml.el]: I find this strange.  We already `(require 'merlin)`.  Why do we have to define this variable explicitly?  Does Merlin not define it?

**Comment by lunaryorn** [comment]
I'm unsure about this `defvar` thing, but other than that it looks good.

But I understand that this is a fairly recent feature in Merlin as well.  Don't we need to update the `Package-Requires` then?  I'd like to make sure that users who install from MELPA Stable get a consistent set of dependencies.

**Comment by def-lkb** [comment]
I agree it's strange. When I tried on my setup, it was *once* ignored by emacs, which considered merlin-command-priority has a normal lexically scoped variable.
Of course there was a reason, I am almost sure I mixed outdated elc files, but just to avoid this fragility, I preferred to repeat the defvar definition, so that the compiler is less sensitive to scope.

This will also be the case for users having a non git version of merlin, but the nice thing is that nothing in the code refers to new definitions, so it's a transparent dropin, the merlin-command-priority will just be ignored by older versions.

**Comment by lunaryorn** [comment]
@def-lkb So flycheck-ocaml will continue to work with older Merlin versions that do not provide this new option?  That's cool, and a good reason to define the variable explicitly.  Thanks!

**Comment by lunaryorn** [comment]
@def-lkb Merged!  Thanks for your help and your patch for this problem, it's greatly appreciated!  And thanks for Merlin, it makes OCaml in Emacs insanely awesome!  It's just a shame that I don't get to do any OCaml anymore :(

**Comment by def-lkb** [comment]
Thank you, well don't despair, you might have other occasions to do OCaml, and merlin will be even more awesome at that point :D.

**Pull request merged**